### PR TITLE
Fixed Warnings (now below 100 again)

### PR DESCRIPTION
**Pull request by daalta**
Similar to https://github.com/unitystation/unitystation/pull/4180 https://github.com/unitystation/unitystation/pull/4099

Fixed some simple compilation warnings. Such as unassigned variables.

GameManager also had some null checks for floats - which caused warnings, because floats cannot be null.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Alien/AlienEggCycle.cs [hunk 1]
```diff
@@ -24,6 +24,7 @@ public class AlienEggCycle : MonoBehaviour, IInteractable<HandApply>, IServerSpa
 		[SerializeField]
 		private EggState initialState = EggState.Growing;
 
+		// TODO This is entirely unused.
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Alien/AlienEggCycle.cs:27]: Should be used on caverna away site unless i missed doing that.

Comment (daalta) [reply UnityProject/Assets/Scripts/Alien/AlienEggCycle.cs:27]: @ThatDan123 It's definitely set inside the editor. But the compiler can't find any references to it being *used* anywhere.

We might want to leave the variable for future use, but are you certain it's implemented?

**Comment by daalta** [comment]
Alrighty, I think I'm done for today. Would be good if we can get this merged today so there's no conflicts.

**Comment by ThatDan123** [comment]
Looks good

**Pull request merged**